api/specs.proto

```diff
@@ -336,6 +334,15 @@ message NetworkSpec {
 	bool ingress = 7;
 }
 
+// NetworkSpec specifies user defined network parameters.
+message NetworkSpec {
+	Annotations annotations = 1 [(gogoproto.nullable) = false];
+
+	oneof spec {
```

Comment:
Yeah, it is a bit stuttery isn't it. Some of the time you omit it, ie. it is `nw.Spec.GetCNM()` and not `nw.Spec.Spec.CNM()`, but for other places (e.g. type switched) you end up with the stuttering.

I had considered `kind` or `type` which seemed okish, brainstorming a bit more now comes up with `backend` or `config` or `backend_config` perhaps. WDYT?